sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BeamBigQuerySqlDialect.java

```diff
@@ -253,6 +259,11 @@ private void unparseTrim(SqlWriter writer, SqlCall call, int leftPrec, int right
     writer.endFunCall(trimFrame);
   }
 
+  private void unparseDateTimeLiteralWrapperFunction(
+      SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    writer.literal(call.operand(0).toString().replace("TIMESTAMP", "DATETIME"));
```

Comment:
I'm not a fan of using `replace` here. I believe the `operand(0)` will be a `SqlTimestampLiteral`? If so, you can call `toFormattedString` instead: https://github.com/apache/calcite/blob/52a57078ba081b24b9d086ed363c715485d1a519/core/src/main/java/org/apache/calcite/sql/SqlTimestampLiteral.java#L54

Reply:
It's a `SqlAbstractDateTimeLiteral`, a super class of `SqlTimestampLiteral`, but we can cast here because we know what the real type is. I am ok with either approach.

Reply:
Ack.

Reply:
Done.